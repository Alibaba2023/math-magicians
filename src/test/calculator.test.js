import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Components/calculator';

describe('Calculator Component', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('handles button click correctly', () => {
    const { getByText, getByDisplayValue } = render(<Calculator />);
    const inputElement = getByDisplayValue('');

    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(inputElement.value).toBe('3');
  });

  test('handles AC button click correctly', () => {
    const { getByText, getByDisplayValue } = render(<Calculator />);
    const inputElement = getByDisplayValue('');

    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('AC'));
    expect(inputElement.value).toBe('');
  });

  test('handles division by zero correctly', () => {
    const { getByText, getByDisplayValue } = render(<Calculator />);
    const inputElement = getByDisplayValue('');

    fireEvent.click(getByText('5'));
    fireEvent.click(getByText('รท'));
    fireEvent.click(getByText('0'));
    fireEvent.click(getByText('='));
    expect(inputElement.value).toBe("Can't divide by 0.");
  });
});
