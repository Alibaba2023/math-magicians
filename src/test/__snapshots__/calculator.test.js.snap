// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component renders correctly 1`] = `
<div
  class="calculator-container"
>
  <input
    class="input-result"
    readonly=""
    type="text"
    value=""
  />
  <ul
    class="numbers-container"
  >
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        AC
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        +/-
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        %
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        รท
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        7
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        8
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        9
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        x
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        4
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        5
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        6
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        -
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        1
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        2
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        3
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        +
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        0
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        .
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        =
      </button>
    </li>
  </ul>
</div>
`;

exports[`renders correctly 1`] = `
<div
  class="calculator-container"
>
  <input
    class="input-result"
    readonly=""
    type="text"
    value=""
  />
  <ul
    class="numbers-container"
  >
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        AC
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        +/-
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        %
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        รท
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        7
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        8
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        9
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        x
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        4
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        5
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        6
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        -
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        1
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        2
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        3
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        +
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        0
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        .
      </button>
    </li>
    <li
      class="list-item"
    >
      <button
        class="number"
        type="button"
      >
        =
      </button>
    </li>
  </ul>
</div>
`;
